ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlAbstractDecoder.kt

```diff
@@ -147,7 +147,11 @@ public abstract class TomlAbstractDecoder : AbstractDecoder() {
         val keyValue = decodeKeyValue()
         try {
             return when (val value = keyValue.value) {
-                is TomlLong -> decodeUnsignedInteger((value.content as Long).toULong(), keyValue.lineNo)
+                is TomlLong -> {
```

Comment:
I first though that exception should be thrown for negative number and TomlUnsignedLong, but then understood the idea. Got it! LGTM.